File: bush.c
//
// Created by alisa on 06.10.2021.
//

#include <string.h>
#include <stdlib.h>
#include "bush.h"

void InRndBush(struct bush *b) {
    b->month = rand() % 12;
}

int InBush(struct bush *b, FILE *ifst) {
    char monthStr[10];
    if (fscanf(ifst, "%s", monthStr) == EOF)
        return EOF;
    if (strcmp(monthStr, "JANUARY") == 0) {
        b->month = JANUARY;
    } else if (strcmp(monthStr, "FEBRUARY") == 0) {
        b->month = FEBRUARY;
    } else if (strcmp(monthStr, "MARCH") == 0) {
        b->month = MARCH;
    } else if (strcmp(monthStr, "APRIL") == 0) {
        b->month = APRIL;
    } else if (strcmp(monthStr, "MAY") == 0) {
        b->month = MAY;
    } else if (strcmp(monthStr, "JUNE") == 0) {
        b->month = JUNE;
    } else if (strcmp(monthStr, "JULY") == 0) {
        b->month = JULY;
    } else if (strcmp(monthStr, "AUGUST") == 0) {
        b->month = AUGUST;
    } else if (strcmp(monthStr, "SEPTEMBER") == 0) {
        b->month = SEPTEMBER;
    } else if (strcmp(monthStr, "OCTOBER") == 0) {
        b->month = OCTOBER;
    } else if (strcmp(monthStr, "NOVEMBER") == 0) {
        b->month = NOVEMBER;
    } else if (strcmp(monthStr, "DECEMBER") == 0) {
        b->month = DECEMBER;
    } else {
        return 0;
    }
    return 1;
}

void OutBush(struct bush *b, FILE *ofst) {
    char *monthStr;
    if (b->month == JANUARY) {
        monthStr = "JANUARY";
    } else if (b->month == FEBRUARY) {
        monthStr = "FEBRUARY";
    } else if (b->month == MARCH) {
        monthStr = "MARCH";
    } else if (b->month == APRIL) {
        monthStr = "APRIL";
    } else if (b->month == MAY) {
        monthStr = "MAY";
    } else if (b->month == JUNE) {
        monthStr = "JUNE";
    } else if (b->month == JULY) {
        monthStr = "JULY";
    } else if (b->month == AUGUST) {
        monthStr = "AUGUST";
    } else if (b->month == SEPTEMBER) {
        monthStr = "SEPTEMBER";
    } else if (b->month == OCTOBER) {
        monthStr = "OCTOBER";
    } else if (b->month == NOVEMBER) {
        monthStr = "NOVEMBER";
    } else if (b->month == DECEMBER) {
        monthStr = "DECEMBER";
    } else {
        monthStr = "Incorrect_month!";
    }
    fprintf(ofst, "It is bush: month = %s.", monthStr);
}

File: bush.py
from plant import *
from enum import Enum


class Month(Enum):
    JANUARY = 1
    FEBRUARY = 2from plant import *
from enum import Enum


class Month(Enum):
    JANUARY = 1
    FEBRUARY = 2
    MARCH = 3
    APRIL = 4
    MAY = 5
    JUNE = 6
    JULY = 7
    AUGUST = 8
    SEPTEMBER = 9
    OCTOBER = 10
    NOVEMBER = 11
    DECEMBER = 12


# ----------------------------------------------
class Bush(Plant):
    def __init__(self):
        super().__init__()
        self.month = Month.JANUARY

    def ReadStrArray(self, strArray, i):
        if i >= len(strArray):
            return 0
        if strArray[i] == "JANUARY":
            self.month = Month.JANUARY
        if strArray[i] == "FEBRUARY":
            self.month = Month.FEBRUARY
        if strArray[i] == "MARCH":
            self.month = Month.MARCH
        if strArray[i] == "APRIL":
            self.month = Month.APRIL
        if strArray[i] == "MAY":
            self.month = Month.MAY
        if strArray[i] == "JUNE":
            self.month = Month.JUNE
        if strArray[i] == "JULY":
            self.month = Month.JULY
        if strArray[i] == "AUGUST":
            self.month = Month.AUGUST
        if strArray[i] == "SEPTEMBER":
            self.month = Month.SEPTEMBER
        if strArray[i] == "OCTOBER":
            self.month = Month.OCTOBER
        if strArray[i] == "NOVEMBER":
            self.month = Month.NOVEMBER
        if strArray[i] == "DECEMBER":
            self.month = Month.DECEMBER
        i += 1
        return i

    def Write(self, ostream, i):
        ostream.write("{}: Bush: month = {}, name = {}, GetQuotient = {}".format \
                          (i, self.month, self.name, self.GetQuotient()))
        pass

    def GetQuotient(self):
        length = len(self.name)
        vowels = 0
        for i in range(length):
            if self.name[i] == 'a' or self.name[i] == 'o' or self.name[i] == 'e' or self.name[i] == 'u' \
                    or self.name[i] == 'i':
                vowels += 1
        return float(vowels / length)
        pass

    MARCH = 3
    APRIL = 4
    MAY = 5
    JUNE = 6
    JULY = 7
    AUGUST = 8
    SEPTEMBER = 9
    OCTOBER = 10
    NOVEMBER = 11
    DECEMBER = 12


# ----------------------------------------------
class Bush(Plant):
    def __init__(self):
        super().__init__()
        self.month = Month.JANUARY

    def ReadStrArray(self, strArray, i):
        if i >= len(strArray):
            return 0
        if strArray[i] == "JANUARY":
            self.month = Month.JANUARY
        if strArray[i] == "FEBRUARY":
            self.month = Month.FEBRUARY
        if strArray[i] == "MARCH":
            self.month = Month.MARCH
        if strArray[i] == "APRIL":
            self.month = Month.APRIL
        if strArray[i] == "MAY":
            self.month = Month.MAY
        if strArray[i] == "JUNE":
            self.month = Month.JUNE
        if strArray[i] == "JULY":
            self.month = Month.JULY
        if strArray[i] == "AUGUST":
            self.month = Month.AUGUST
        if strArray[i] == "SEPTEMBER":
            self.month = Month.SEPTEMBER
        if strArray[i] == "OCTOBER":
            self.month = Month.OCTOBER
        if strArray[i] == "NOVEMBER":
            self.month = Month.NOVEMBER
        if strArray[i] == "DECEMBER":
            self.month = Month.DECEMBER
        i += 1
        return i

    def Write(self, ostream, i):
        ostream.write("{}: Bush: month = {}, name = {}, GetQuotient = {}".format \
                          (i, self.month, self.name, self.GetQuotient()))
        pass

    def GetQuotient(self):
        length = len(self.name)
        vowels = 0
        for i in range(length):
            if self.name[i] == 'a' or self.name[i] == 'o' or self.name[i] == 'e' or self.name[i] == 'u' \
                    or self.name[i] == 'i':
                vowels += 1
        return float(vowels / length)
        pass
